segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -144,11 +147,29 @@
                             CompletableFuture<Void> f;
                             if (chunksToConcat.size() > 1) {
                                 // Concat
-                                f = concatChunks();
+                                f = concatChunks()
+                                .handleAsync((vv, ex) -> {
+                                    if (null != ex) {
+                                        ex = Exceptions.unwrap(ex);
+                                        if (ex instanceof InvalidOffsetException) {
+                                            // Skip ahead by 1 chunk.
+                                            targetChunkName = chunksToConcat.get(1).getName();
+                                            chunksToConcat.clear();
+                                            skipFailed.set(true);
+                                            return null;
+                                        }
+                                        throw new CompletionException(ex);
+                                    }
+                                    return v;
+                                }, chunkedSegmentStorage.getExecutor());
                             } else {
                                 f = CompletableFuture.completedFuture(null);
                             }
                             return f.thenRunAsync(() -> {
```

Comment:
You do not need `Async` here

Reply:
leaving it as it is - keeping the old pattern.